src/Application/Injectors/Configuration.php

```diff
@@ -41,6 +41,7 @@ public function __invoke(): array
                     $config = require $configPath;
                 }
 
+                $config = ConfigResolver::mergeInputRequirements($config, $input);
                 return ConfigResolver::resolve($config, DirectoryResolver::resolve($input));
```

Comment:
We could consider moving all of this into the `resolve` method, and just make the resolve method take the following
```
ConfigResolver::resolve($config, $input);
```
Then inside resolve make it have the two lines, as `resolve` now has the `$input`.
```
ConfigResolver::mergeInputRequirements($config, $input)
DirectoryResolver::resolve($input)
```

Reply:
@olivernybroe I've got a few things I would like to clarify before going further.

1) $config passed just because base config was resolved and included outside of resolve method. In case we would like to pass input directly it seems to be excess to have config option and I suppose we can get rid of it since class marked as internal and keeping an API isn't required. Am I missing some use case?

2)  We have a call to resolve in both  InsightsCommand and AnalyseCommand. But check for resolved path totally different. Can we make it common?

Reply:
1. Just so I am sure we are talking about the same thing, you are talking about that we do `ConfigResolver::resolvePath` first and then `require` that path, as that is our config file? If that is the case, I can see what you mean, by moving that, we can take the `input` arg as the only one. However just to come with two things right of my head of why it was done this way. In tests, we would like to resolve the config from an array, and not having to create a file for it, that we can test with. Another thing is that in the Laravel command we require the config file to be there, which we don't do normally. 
It would be cool with a nice way to clean that part up and also remove the duplication there is because of this. However that change could be in another PR, not something that is required for this PR.

2. As said before, would love to have a common solution for that ❤️ 
That might be a great solution, as it allows the command itself to handle what to do, or do anything at all.


@devyk To wrap this up. 
Let's not make the change I suggested here and just start with the solution you did originally. If you are up for it, then we could create a separate PR, which cleans it up and at the same time handles the other issues/suggestions you mentioned.